deep_quoridor/src/main.py

```diff
@@ -9,42 +9,20 @@
 board = game.render()
 print(board)
 
-# Manually select moves (modify these to test different scenarios)
-actions = [
-    # 0- Move forward
-    game.action_params_to_index(1, 4, 0),
-    # 1- Move forward (opponent)
-    game.action_params_to_index(7, 4, 0),
-    # 0- Add a Horizontal wall
-    game.action_params_to_index(6, 4, 2),
-    # 1- Add a vertical wall
-    game.action_params_to_index(0, 3, 1),
-    # 0- Place next to the opponent, let it jump
-    game.action_params_to_index(7, 3, 0),
-    # 1- Place a wall behind me so that the P0 can't jump
-    game.action_params_to_index(7, 4, 1),
-]  # Example moves for testing
-
 step = 0
 for agent in game.agent_iter():
     observation, reward, termination, truncation, info = game.last()
 
     mask = observation["action_mask"]
-    print(f"Valid moves for agent {agent}:")
-    for i in range(game.board_size**2):  # For now only showing moves
-        if mask[i] == 1:
-            print(f"{i}: {game.action_index_to_params(i)}")
-
-    # End of hardcoded test actions
-    if len(actions) == 0:
-        break
 
     if termination or truncation:
         action = None
-        print(f"\nGame Over! {agent} wins.")
+        winner = max(game.rewards, key=game.rewards.get)
+        print(f"\nGame Over! {winner} wins.")
+        break
     else:
         # Hardcoded actions for now
-        action = actions.pop(0)
+        action = game.action_space(agent).sample(mask)
```

Comment:
We have a baseline agent!

Reply:
Is `.sample()` uniform random?

Reply:
Apparently yes in this case: https://gymnasium.farama.org/api/spaces/#gymnasium.spaces.Space.sample

Reply:
We should write a better heuristic, e.g. follow the shortest path to the target if our path is shorter, or place a wall in the path of the shorter path of the adversary otherwise

Reply:
I'll try that tonight